### ISSUE TITLE
[Bug] Exception error_code is "access_denied" when user clicks cancel button on embedded account picker

### ISSUE THREAD

**Issue by riyabans** [AzureAD/microsoft-authentication-library-for-dotnet#3055]
**Logs and network traces**
Without logs or traces, it is unlikely that the team can investigate your issue. Capturing logs and network traces is described in [Logging wiki](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/logging).

**Which version of MSAL.NET are you using?**
MSAL.NET 4.37.0.0 (Also verified issue persists on 4.39.0.0)

**Platform**
.NET 4.7, WPF

**What authentication flow has the issue?**
* Desktop / Mobile
    * [ ] Interactive
    * [x] Integrated Windows Authentication
    * [ ] Username Password
    * [ ] Device code flow (browserless)

**Is this a new or existing app?**
The app is in production, I haven't upgraded MSAL, but started seeing this issue.

**Repro**
1. Use Embedded account picker, interactive auth sign in.
2. Use a MSA Account.
3. On clicking the account credentials, we get redirected to screen "Are you trying to sign in to Microsoft Whiteboard Services?"
4. Click the cancel button here.

**Expected behavior**
The exception thrown by this event should have some info about user_canceled/authentication_canceled event. 

**Actual behavior**
The exception thrown by this event is:
- $exception {"The user has denied access to the scope requested by the client application."} Microsoft.Identity.Client.MsalServiceException
ErrorCode "access_denied" string

**Possible solution**
If we can add a sub error code for this event, then the fix for below issue might work here:

**Additional context / logs / screenshots / links to code**
![image](https://user-images.githubusercontent.com/95686314/145015111-49788a6c-fd66-4c0c-a691-0b88ae431b53.png)

**Comment by bgavrilMS** [comment]
Is this using embedded browser or WAM? Do you have `.WithBroker` anywhere?

**Comment by riyabans** [comment]
Using embedded browser.
Redirect URI: "ms-appx-web://microsoft.aad.brokerplugin/"
WithWindowsBroker(false)

**Comment by bgavrilMS** [comment]
Agreed, this screen is pretty much asking for consent for "User.Read" scope, but in a more "friendly" way that the typical consent screen. If user hits cancel, the standard error code sent by AAD is "access_denied", which means "user denied the application to access content on its behalf". This is different from "user cancelled authentication".
